common/compiler/solidity.go

```diff
@@ -110,95 +111,82 @@ func (sol *Solidity) Version() string {
 	return sol.version
 }
 
-func (sol *Solidity) Compile(source string) (contracts map[string]*Contract, err error) {
-
+// Compile builds and returns all the contracts contained within a source string.
+func (sol *Solidity) Compile(source string) (map[string]*Contract, error) {
+	// Short circuit if no source code was sepcified
 	if len(source) == 0 {
-		err = fmt.Errorf("empty source")
-		return
+		return nil, errors.New("solc: empty source string")
 	}
-
+	// Create a safe place to dump compilation output
 	wd, err := ioutil.TempDir("", "solc")
 	if err != nil {
-		return
+		return nil, fmt.Errorf("solc: failed to create temporary build folder: %v", err)
 	}
 	defer os.RemoveAll(wd)
 
-	in := strings.NewReader(source)
-	var out bytes.Buffer
-	// cwd set to temp dir
+	// Assemble the compiler command, change to the temp folder and capture any errors
+	stderr := new(bytes.Buffer)
+
 	cmd := exec.Command(sol.solcPath, params...)
 	cmd.Dir = wd
-	cmd.Stdin = in
-	cmd.Stdout = &out
-	err = cmd.Run()
-	if err != nil {
-		err = fmt.Errorf("solc error: %v", err)
-		return
-	}
+	cmd.Stdin = strings.NewReader(source)
+	cmd.Stderr = stderr
 
+	if err := cmd.Run(); err != nil {
+		return nil, fmt.Errorf("solc: %v\n%s", err, string(stderr.Bytes()))
+	}
+	// Sanity check that something was actually built
 	matches, _ := filepath.Glob(wd + "/*.binary")
 	if len(matches) < 1 {
-		err = fmt.Errorf("solc error: missing code output")
-		return
+		return nil, fmt.Errorf("solc: no build results found")
 	}
-
-	contracts = make(map[string]*Contract)
+	// Complilation succeeded, assemble and return the contracts
```

Comment:
I always use full sentences as variable, fumction, etc declarations. Within
the code, I never add the dots. It's a personal style :D
On Aug 25, 2015 10:55 PM, "Felix Lange" <notifications@github.com> wrote:

> In common/compiler/solidity.go
> <https://github.com/ethereum/go-ethereum/pull/1717#discussion_r37911317>:
>
> >  	}
> > -
> > -	contracts = make(map[string]*Contract)
> > +	// Complilation succeeded, assemble and return the contracts
>
> // Compilation succeeded, assemble and return the contracts.
>
> Sentences. End. With. Period.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ethereum/go-ethereum/pull/1717/files#r37911317>.
>